components/dashboard/src/start/VerifyModal.tsx

```diff
@@ -0,0 +1,220 @@
+                    },
+                });
+            }
+            return verified;
+        };
+
+        const reset = () => {
+            setState({
+                ...state,
+                sent: undefined,
+                message: undefined,
+                token: undefined,
+            });
+        };
+        return (
+            <Modal
+                onClose={() => {}}
+                closeable={false}
+                onEnter={verifyToken}
+                title="User Validation Required"
+                buttons={
+                    <div>
+                        <button className="ml-2" disabled={!isTokenFilled()} onClick={verifyToken}>
+                            Validate Account
+                        </button>
+                    </div>
+                }
+                visible={true}
+            >
+                <Alert type="warning" className="mt-2">
+                    To use Gitpod you'll need to validate your account with your phone number. This is required to
+                    discourage and reduce abuse on Gitpod infrastructure.
+                </Alert>
+                <div className="pt-4">
+                    <button className="gp-link" onClick={reset}>
+                        &larr; Use a different phone number
+                    </button>
+                </div>
+                <div className="pt-4">
+                    Enter the verification code we sent to {state.phoneNumber}.<br />
+                    Having trouble?{" "}
+                    <a className="gp-link" href="https://www.gitpod.io/contact/support">
+                        Contact support
+                    </a>
+                </div>
+                {state.message ? (
+                    <Alert type={state.message.type} className="mt-4 py-3">
+                        {state.message.text}
+                    </Alert>
+                ) : (
+                    <></>
+                )}
+                <div className="mt-4">
+                    <h4>Verification Code</h4>
+                    <input
+                        autoFocus={true}
+                        className="w-full"
+                        type="text"
+                        placeholder="Enter code sent via SMS"
+                        onChange={(v) => {
+                            setState({
+                                ...state,
+                                token: v.currentTarget.value,
+                            });
+                        }}
+                    />
+                </div>
+            </Modal>
+        );
+    } else {
+        const continueStartWorkspace = () => {
+            window.location.reload();
+            return true;
+        };
+        return (
+            <Modal
+                onClose={continueStartWorkspace}
+                closeable={false}
+                onEnter={continueStartWorkspace}
+                title="User Validation Successful"
```

Comment:
**thought**: In case this could be interesting we could surface the account validation status on the account settings.

Reply:
It's a SaaS-only concern and should really touch as few places as possible. Why would it be interesting to show it there?

Reply:
Good point! Not much interesting, indeed.

I probably had in mind how we could use the same flow to validate with a credit card if we need in the future so that users benefit from having a credit card in place when and if they enable billing for their personal account. Resolving for now. 💳

Reply:
We can mark users as verified when they take other actions that make them trustworthy. No UI needed.

Reply:
Yeah, maybe also something to consider surfacing in the admin area. It will be interesting to see if users manage to bypass somehow this verification process for abusing the infrastructure. 💡 

Cc @jldec because https://github.com/gitpod-io/gitpod/issues/7597
Cc @geropl because https://github.com/gitpod-io/gitpod/issues/11469